enigma-tools-u/src/web3_utils/enigma_contract.rs

```diff
@@ -88,35 +73,53 @@ pub trait ContractFuncs<G> {
 
 impl<G: Into<U256>> ContractFuncs<G> for EnigmaContract {
     #[logfn(DEBUG)]
-    fn register(&self, signing_address: H160, _report: String, _signature: String, gas: G, confirmations: usize) -> Result<TransactionReceipt, Error> {
+    fn register(&self, signing_address: H160, report: String, signature: String, gas: G, confirmations: usize) -> Result<TransactionReceipt, Error> {
         // register
         let mut opts = Options::default();
         opts.gas = Some(gas.into());
         // call the register function
-        let report = _report.as_bytes().to_vec();
-        let signature = _signature.from_hex()?;
-//        println!("The report signer: {:?}", signing_address);
-//        println!("The report: {}", str::from_utf8(&report).unwrap());
-//        println!("The report signature: {}", signature.to_hex());
-        let call = self.w3_contract.call_with_confirmations("register", (signing_address, report, signature), self.account, opts, confirmations);
-        let receipt = match call.wait() {
-            Ok(receipt) => receipt,
-            Err(e) => {
-                return Err(errors::Web3Error { message: format!("Unable to call register: {:?}", e) }.into());
-            }
-        };
+        let report = report.as_bytes().to_vec();
+        let signature = signature.from_hex()?;
+        let call = signed_call_with_confirmations(
+            &self.web3,
+            &self.ethabi_contract,
+            self.w3_contract.address(),
+            H256([0; 32]), // TODO set to a real key
```

Comment:
make sure not to push this to develop! (I know you are working on this)

Reply:
Thanks :)